apply_kernel.json

```diff
@@ -21,18 +21,22 @@
             "required": true
         },
         "kernel": {
-            "description": "The kernel to be applied on the data cube. The kernel has to be as many dimensions as the data cube has dimensions.",
+            "description": "A two-dimensional weighted kernel. Each dimension of the kernel must have an uneven number of elements, otherwise the process throws a `KernelDimensionsUneven` error.",
             "schema": {
+                "description": "A two-dimensional array of numbers.",
                 "type": "array",
                 "subtype": "kernel",
                 "items": {
-                    "description": "Usually a multi-dimensional array of numbers."
+                    "type": "array",
+                    "items": {
+                        "type": "number"
+                    }
                 }
             },
             "required": true
         },
         "factor": {
-            "description": "A factor that is multiplied to each value computed by the focal operation.\n\nThis is basically a shortcut for explicitly multiplying each value by a factor afterwards, which is often required for some kernel-based algorithms such as the Gaussian blur.",
+            "description": "A factor that is multiplied to each value after the kernel has been applied.\n\nThis is basically a shortcut for explicitly multiplying each value by a factor afterwards, which is often required for some kernel-based algorithms such as the Gaussian blur.",
```

Comment:
>  after the kernel has been applied

does this have to be specified explicitly here? Isn't that just an implementation detail? e.g. a backend may want to pre-multiply the kernel before applying the convolution instead of post-multiplying the output.

Reply:
This is a description for the user. I have a hard time describing this in a implementation agnostic way. Does it really matter how implementations do it internally? The result should be the same anyway... Writing something like "A factor that is multiplied either to each computed value afterwards or to each kernel element beforehand." is confusing for the user, I think. Suggestions?

Reply:
I would just avoid explicit "afterwards/beforehand". Suggestion:

> An additional multiplication factor to multiply the result with

Reply:
Okay, I'll remove that.

Reply:
But it's still mentioned in the process description and I really find it difficult to explain how the process works without saying whether the factor is applied to each value of the kernel or the result of the focal operation.